bob-api/api/de_onderwijzers_api.php

```diff
@@ -0,0 +1,195 @@
+ * - https://graphql.org/learn/queries
+ * - https://graphql.org/learn/queries/#variables
+ *
+ * Als je debug logging aan wilt hebben, zet dan $DEBUG_ENABLE naar true
+ *
+ * GraphiQl docs: http://api-test.edwh.nl/graphql/?api_key=IBFiOq8Fyoxcqxo4gyDT7pBXzC0REahqcOXu1Nql
+ *
+ * @package api
+ */
+class DeOnderwijzers
+{
+    public static $DEBUG_ENABLED = false;
+
+    /**
+     * The API key to use during the connection
+     */
+    const API_KEY = "IBFiOq8Fyoxcqxo4gyDT7pBXzC0REahqcOXu1Nql";
+
+    /**
+     * The base url of the external connection
+     */
+    const BASE_URL = "http://api-test.edwh.nl/graphql/?api_key=";
+
+    /**
+     * Session token for the application. This should only be used for non-user specific actions.
+     * Once we have a user, his/her sessionToken should be used for subsequent requests.
+     */
+    private $applicationSessionToken;
+
+    /**
+     * DeOnderwijzers constructor.
+     *
+     * Create a new session token that will be used for all subsequent requests
+     * @throws ErrorException
+     */
+    public function __construct()
+    {
+        $this->applicationSessionToken = $this->generateSessionToken();
+    }
+
+    /**
+     * Get information about the currently logged in user
+     * @param string $sessionToken Session token for a user
+     * @return array
+     * @throws ErrorException
+     */
+    public function getMe(string $sessionToken): array
+    {
+        $this->logDebug("Executing getMe() query");
+
+        $query = <<<GRAPHQL
+        query RootQuery(\$sessionToken: UUID) {
+            viewer(sessionToken: \$sessionToken) {
+                me {
+                  id,
+                  email,
+                  apiToken,
+                  avatar
+                }
+            }
+        }
+GRAPHQL;
+
+        $result = $this->executeQuery($query, array("sessionToken" => $sessionToken));
+
+        $this->logDebug("Execution of getMe() finished");
+
+        return $result['data'];
+    }
+
+    /**
+     * Validate the given credentials
+     * @param string $email The email of the user
+     * @param string $password The password of the user
+     * @return array
+     * @throws ErrorException
+     */
+    public function validateCredentials(string $email, string $password): array
+    {
+        $this->logDebug("Executing validateCredentials() query");
+
+        $query = <<<GRAPHQL
+        query RootQuery(\$sessionToken: UUID, \$email: String, \$passwordHash: String, \$hardware: JSONString) {
+           viewer(sessionToken: \$sessionToken) {
+            validateCredentials(email: \$email, passwordHash: \$passwordHash, hardware: \$hardware) {
+              ok,
+              feedback,
+              user {
+                id,
+                email,
+                name,
+                permalink,
+                avatar
+              },
+              sessionToken
+            }
+          }
+        }
+GRAPHQL;
+
+        $passwordHash = md5($password);
```

Comment:
Hee Bob, volgens mij kan je ook beter een salt gebruiken als je een wachtwoord met md5 wilt encryten. Md5 is namelijk vulnerable tegen een rainbow table attack en kan dus vrij snel gekraakt worden met een goede wordlist en hashcat

Reply:
*encrypten

Reply:
thanks voor de tip, pak ik op!